types/types.go

```diff
@@ -547,9 +547,35 @@ type SystemContext struct {
 	CompressionLevel *int
 }
 
+// ProgressEvent is the type of events a progress reader can produce
```

Comment:
```suggestion
// ProgressEvent is the type of events a progress reader can produce
// Warning: new event types may be added any time.
```
(… although, right now, we can’t really add new events that don’t set `Artifact`+`Offset` without breaking users. Eventually this should probably be a complex typed API that provides only the right fields for the right events, but not now — this is great because it keeps existing consumers working.)

Reply:
Alright, I will apply the suggestion :+1: